src/formatDate.js

```diff
@@ -50,7 +50,35 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const dateObject = {};
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = dateArray[i];
+  }
+
+  for (const element of toFormat) {
+    for (const key in dateObject) {
+      if (element === key) {
+        result.push(dateObject[key]);
+      }
+
+      if (element === 'YY' && key === 'YYYY') {
```

Comment:
Consider using variables with meaningful names instead of magic strings
```suggestion
      if (element === SHORT_YEAR && key === LONG_YEAR) {
```